### PR TITLE
AsteroidStation

### PR DESCRIPTION
**Pull request by ghost**
Copy paste from [Frozen's PR](https://github.com/HippieStation/HippieStation/pull/7485), fixed conflicts: 
Since the default hippiestation map is getting boring. I've decided to add the revived AsteroidStation that /tg/ sadly closed.

Credits to @ CorruptComputer for the entire, original map!

:cl: CorruptComputer, FrozenGuy5
add: Adds AsteroidStation!
/:cl:

THINGS TO DO:
arrivals is broken
i think some areas aren't properly specified
escape is probably broken
it only has SINGULO and idk if it can power the station ever since the nerf
probably not techwebs compliant at all
i think there might be issues with ore redempetion

### CODE REVIEW

**Comment by ghost** [comment]
You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by JohnGinnane** [comment]
Original PR: https://github.com/tgstation/tgstation/pull/33164

**Comment by ghost** [comment]
@JohnGinnane https://cdn.discordapp.com/attachments/129145765049139200/385290291181256704/AsteroidStation.png from https://github.com/tgstation/tgstation/pull/33164

**Comment by JohnGinnane** [comment]
damn this looks like a LOT of fun

**Comment by ghost** [comment]
more like assteroid haha

**Comment by ghost** [comment]
lots of broken paths [DNM]

**Comment by TrickyDickRichard** [comment]
i dont see a ghettochem and u gotta add forge but looks fun

**Comment by ghost** [comment]
there's bugs to be fixed most importantly

**Comment by ghost** [comment]
arrivals is broken
i think some areas aren't properly specified
escape is probably broken
it only has SINGULO and idk if it can power the station ever since the nerf
probably not techwebs compliant at all

gonna leave it for now and probably continue touching it up on sunday

**Comment by Jujumatic** [comment]
we could always just add a tesla setup with the singulo setup like we have in the map we currently use

**Comment by praisenarsie** [comment]
that image is slightly outdated

anyway the major things that need to be done is
removing shuttles & replacing with templates
adding techwebs
make the map somehow smaller
lOOOOOONG HALLWAYS FUCK EM WE NEED TO KILL THEM SOMEHOW
where the fuck is mining shuttle (also we should consider adding the OLD asteroid back instead of lavaland for old time's sake maybe)

the less major things is
add pool
remove that riveted wall in the toxins dome
replace renault with hedgehog
add an incinerator to the atmos burn room OR move toxins there OR leave it
plasma window some shit
add EXPERIMENTOR
add forge

File: config/maps.txt [hunk 1]
```diff
@@ -39,3 +39,7 @@ endmap
 
 map runtimestation
 endmap
+
+map AsteroidStation
```
Comment (praisenarsie) [inline config/maps.txt]: decapitalize this for consistency

**Comment by CorruptComputer** [comment]
I've actually been wanting to continue this map, don't know if someone else wanted to maintain, but I'm definitely interested. I would need to work a bit and play on this codebase though to get up to speed with the differences.

**Comment by JohnGinnane** [comment]
I think if you made the map /tg/ compatible it would probably almost entirely be compatible here too, because we're just downstream from /tg/ and try to keep up to date with them

**Comment by CorruptComputer** [comment]
Alright, as for the broken things I will be working on all those next week. I recently decided to redo all the pipes and wiring to remove the departmental checkpoints for them, as they were pretty poorly received.

**Comment by ghost** [comment]
do you have access to our discord? we'd probably be more able to give a helping hand if you are

**Comment by CorruptComputer** [comment]
Nope, if you want to add me I'm CorruptComputer# 0001

**Comment by earthmoonjune** [comment]
boxstation has gotten stale after all this time and i've always thought a map rotation would be neat. +1

**Comment by CorruptComputer** [comment]
@McBawbaggings I was gonna take a look at this over the weekend, could I get an invite to that Discord server?

**Comment by tortellinitony** [comment]
There must be a pool, mime office, and clown office. Also fix the bugs. I don't think this is viable, but if you're willing to put in the work then I'd like to see how this turns out.

**Comment by praisenarsie** [comment]
lack motivation to keep working on this

**Comment by CorruptComputer** [comment]
Could this be re-opened? It is still being worked on by myself and @McBawbaggings

**Comment by yoyobatty** [comment]
You can always open a new pull request

**Comment by yoyobatty** [comment]
It looks great but I personally think it needs a few more aesthetic tweaks to give it some character. If you need some reference, meta/delta/omega station have some great ideas 👍

**Comment by JohnGinnane** [comment]
@yoyobatty I think adding some detail is something that can happen over time. Like to say it needs some character is vague, so I don't think it's high on the priority list

Any cool stuff like pool or w/e can be added when it's functional and stuff but it's up to CorruptComputer

**Comment by CorruptComputer** [comment]
Many of the things requested here have already been added, however I do agree that more detail does need to be added. Along with a rigorous playtest to ensure I haven't broken anything. I do test all my changes, however I don't know how to play every job so I can only test the basics in most cases.

**Comment by JohnGinnane** [comment]
I mean if something is really fecked then speedmerge could be done anyway, so it's no biggie

**Comment by yoyobatty** [comment]
So what I mean is that, it needs some visual eye candy in addition to functional features that add some depth, variety and add on to the overall aesthetic such grilles/fuel/water tanks and maint loot. A lot of the hallways and areas currently resemble 4 sided shapes and I mean that to point out the lack of movement variety required for players. 

Edit, looks like I was peaking out an old screenshot, ignore.

**Comment by CorruptComputer** [comment]
Screenshot from newest version: https://cdn.discordapp.com/attachments/391469396482326528/440661222854098945/AsteroidStation.png

**Comment by JohnGinnane** [comment]
image

Only one cloning pod for genetics?

**Comment by TrickyDickRichard** [comment]
if someone could point me to this map's forge, i'd appreciate it because my search came up empty

**Comment by CorruptComputer** [comment]
I looked on the Hippie map for the forge and it's not there. I even searched using the area, no results found. If you could give me an example to work off of I could make it happen!

**Comment by TrickyDickRichard** [comment]
It's west of telecomms and east of the maint surgery room, here's a picture
image

**Comment by CorruptComputer** [comment]
@JohnGinnane Added a second cloning pod. Medical has been really neglected on AstroidStation in the past and has not really changed in a LOOOOONG time, so I have been planning a revamp on it. Gonna try to do that before this gets merged.

**Comment by praisenarsie** [comment]
also, try and remove or revamp the long hallways on the map because if you see a correlation with all the other removed /tg/ maps, they have too much long hallways. you could probably remedy this by adding jumppads, transit pods, a nEW ZlEVEl or just make it circular and accidently create donutstation

**Comment by Jujumatic** [comment]
how's progress going on this?

**Comment by Carbonhell** [comment]
news?

**Comment by CorruptComputer** [comment]
Screenshot of current changes: 
image

The tool I used didn't render the Hippie stuff for some reason, but it should be enough to get an idea of whats changed.

**Comment by Carbonhell** [comment]
so uh,what's missing?is this stale?

**Comment by CorruptComputer** [comment]
@Carbonhell I handed off the map to @McBawbaggings because I have no idea how to make disposals work, and there also seems to be merge conflicts now...

**Comment by yoyobatty** [comment]
If you need help with this, feel free to send a final draft of it over to me and I'll go through mine and tg's checklist.

File: code/game/machinery/computer/teleporter.dm [hunk 2]
```diff
@@ -122,6 +122,8 @@
 	if(regime_set == "Teleporter")
 		for(var/obj/item/device/beacon/R in GLOB.teleportbeacons)
 			if(is_eligible(R) && R.channel == channel)
+		for(var/obj/item/beacon/R in GLOB.teleportbeacons)
```
Comment (CorruptComputer) [inline code/game/machinery/computer/teleporter.dm]: This should be merged with the `for` above, and for the map to work correctly it it needs the extra check for:
 `R.channel == channel`

File: hippiestation/code/game/area/Space_Station_13_areas.dm [hunk 3]
```diff
@@ -67,4 +67,5 @@
 	name = "Security Teleporter Room"
 	icon_state = "teleporter"
 	music = "signal"
-	ambientsounds = ENGINEERING
\ No newline at end of file
+	ambientsounds = ENGINEERING
+	icon_state = "green"
```
Comment (CorruptComputer) [inline hippiestation/code/game/area/Space_Station_13_areas.dm]: Also not quite sure about this being added...?

**Comment by Cyprex** [comment]
Cargo needs bounty consoles(https://github.com/tgstation/tgstation/pull/37833).
Teleporter next to AI entry uses a standard airlock, should probably be command
Where's the xenobio slime killroom?

Some observations:
- Malf AIs might have a harder time defending with the core layout. The back entrance from maintenance definitely seems odd.
- Xenobio: 5 pens in  + 1 that requires some building to set up. Compact setup, less camera movement necessary as a result.
- Toxins testing range seems rather close to the ghetto bar, has this been tested to be safe with maxcaps?
- Are hardsuit storages showing up correctly?
5 Hardsuit storages in engineering, 2 in atmos, 4 in aux base construction, 6 in cargo, 7 in EVA, certainly an abundance, although that's not necessarily a bad thing.
- Atmos layout is definitely different from meta, seems more barebones but atmos is slightly different on every map.
- Maintenance access + getting past one rwall is enough to get into tcomms, One airlock+rwindow would also work.
- Maintenance is spacious and 2-3 tiles wide in some places, generally on the outside of the station and more cramped in others.
- Toxins mixing looks odd on the screenshot.
- What's up with the vault airlocks?

**Comment by Cyprex** [comment]
Departmental Lathes are also necessary, and a circuit imprinter in engineering, preferably in a way that allows atmos techs to access it.

**Comment by Carbonhell** [comment]
Good feedback, thank you very much for your input @Cyprex 
also yeah if you guys need any help with the map i'll be available on the discord for any question

**Comment by Pyko1** [comment]
pool seems to be missing its machinery

**Comment by CorruptComputer** [comment]
@Cyprex 
> Malf AIs might have a harder time defending with the core layout. The back entrance from maintenance definitely seems odd.

I added that because it made sense to me at the time, but it shouldn't be any easier as you have to make it to the AI slot to deactivate them. Which would be very difficult in my experience.

> Xenobio: 5 pens in + 1 that requires some building to set up. Compact setup, less camera movement necessary as a result.

That's just how it was before I started updating it, I can add those soon(tm) as I am still waiting to hear back from McBaw about the disposals.

> Toxins testing range seems rather close to the ghetto bar, has this been tested to be safe with maxcaps?

I did test it with a bomb, and it was fine, haven't checked maxcap though, may be a little close for that. I'll move it down a bit when I get a chance.

> Are hardsuit storages showing up correctly?

Yes.

> 5 Hardsuit storages in engineering, 2 in atmos, 4 in aux base construction, 6 in cargo, 7 in EVA, certainly an abundance, although that's not necessarily a bad thing.

The map is made for a high population, I tried to make it so anyone who would need one would have one.

> Atmos layout is definitely different from meta, seems more barebones but atmos is slightly different on every map.

Yeah, I actually copied this atmos back in 2015 from Meta, made changes to make it work better here of course. But I've been thinking of changing it as this layout is difficult to work with mapping wise.

> Maintenance access + getting past one rwall is enough to get into tcomms, One airlock+rwindow would also work.

Its right next to sec, but yeah I see what you mean. Box is the same way though: 
image

> Maintenance is spacious and 2-3 tiles wide in some places, generally on the outside of the station and more cramped in others.

I suck at creating chaos, and tried to work on this, but maint has always been AsteroidStations weak point.

> Toxins mixing looks odd on the screenshot.

Looks and works fine in game, issue with the screenshot rendering program.

> What's up with the vault airlocks?

Same as above.

> Departmental Lathes are also necessary, and a circuit imprinter in engineering, preferably in a way that allows atmos techs to access it.

No idea this was a thing, haven't played in a veeeeery long time and just now getting back into it.

@Pyko1 
> pool seems to be missing its machinery

They are there, but as I stated none of the Hippie specific stuff rendered with the program.

**Comment by CorruptComputer** [comment]
> 5 Hardsuit storages in engineering, 2 in atmos, 4 in aux base construction, 6 in cargo, 7 in EVA, certainly an abundance, although that's not necessarily a bad thing.

Also wait, there are 5 in atmos and none is cargo. I think you meant mining? Sec also has 2.

**Comment by Cyprex** [comment]
Yeah, science got overhauled, https://github.com/tgstation/tgstation/pull/31026

**Comment by Cyprex** [comment]
If you seek to install a supermatter I'd recommend looking at box station's, although having the air alarm for the SM core chamber not behind an airlock like on meta would be great. Feel free to post screenshots or such if you have a draft.

**Comment by ghost** [comment]
i want this to be real...

**Comment by Carbonhell** [comment]
This seems stale, sadly... If you ever wanna work on it again just ask and i will reopen it

**Comment by CorruptComputer** [comment]
I think I'll be reopening this under my name, there were too many issues with coordination between us to make it work well for us. Right now I am on vacation, however next week I'll be back on it!

**Comment by Carbonhell** [comment]
Fantastic, thank you very much!

**Comment by ghost** [comment]
yay

**Comment by earthmoonjune** [comment]
poggers

**Pull request closed without merging**